airbyte-integrations/connectors/destination-motherduck/destination_motherduck/processors/duckdb.py

```diff
@@ -101,6 +103,46 @@ class DuckDBSqlProcessor(SqlProcessorBase):
     supports_merge_insert = False
     sql_config: DuckDBConfig
 
+    def _execute_sql(self, sql: str | TextClause | Executable) -> Sequence[Any]:
+        """Execute the given SQL statement."""
+        if isinstance(sql, str):
+            sql = text(sql)
+
+        with self.get_sql_connection() as conn:
+            try:
+                result = conn.execute(sql)
+            except (
+                ProgrammingError,
+                SQLAlchemyError,
+            ) as ex:
+                msg = f"Error when executing SQL:\n{sql}\n{type(ex).__name__}{ex!s}"
+                raise SQLRuntimeError(msg) from None  # from ex
+
+            return result.fetchall()
+
+    def _execute_sql_with_buffer(self, sql: str | TextClause | Executable, buffer_data: pa.Table | None) -> Sequence[Any]:
+        """
+        Execute the given SQL statement.
+
+        Explicitly register a buffer table to read from.
+        """
+        if isinstance(sql, str):
+            sql = text(sql)
+
+        with self.get_sql_connection() as conn:
+            try:
+                # This table will now be queryable from DuckDB under the name BUFFER_TABLE_NAME
```

Comment:
@guenp - Is this a permanent table addition or just for the session/query context? Assuming the latter, is there any risk of name collisions? I assume not, but just want to double check.

Reply:
Great question. I believe it's only for the context!